ShittyLINQ/Zip.cs

```diff
@@ -0,0 +1,31 @@
+﻿using System;
+using System.Collections.Generic;
+
+namespace ShittyLINQ
+{
+    public static partial class Extensions
+    {
+        /// <summary>
+        /// <para>
+        /// Combines two Enumerable objects into a sequence of Tuples containing each element
+        /// of the source Enumerable in the first position with the element that has the same
+        /// index in the second Enumerable in the second position.
+        /// </para>
+        /// <para>
+        /// The output sequence will be as long as the shortest input sequence.
+        /// </para>
+        /// </summary>
+        public static IEnumerable<Tuple<T, U>> Zip<T, U>(this IEnumerable<T> source, IEnumerable<U> second)
+        {
+            if (source == null || second == null) throw new ArgumentNullException();
+
+            var firstEnumerator = source.GetEnumerator();
+            var secondEnumerator = second.GetEnumerator();
+
+            while(firstEnumerator.MoveNext() && secondEnumerator.MoveNext())
+            {
+                yield return Tuple.Create(firstEnumerator.Current, secondEnumerator.Current);
```

Comment:
Instead of having to say `Tuple.Create(first, second)` in C# 7 you can say `(first, second)`.

Reply:
I was trying to get this to work but I was getting tons of compiler errors. Does something need to be done to the project in order for it to support C# 7? I just realized I hadn't updated VS2017 since the day it came out.